openedx/core/lib/studio_entities.py

```diff
@@ -0,0 +1,64 @@
+"""Studio entity plugin manager and API."""
+import abc
+
+from openedx.core.lib.api.plugins import PluginManager
+
+
+class StudioEntityPluginManager(PluginManager):
+    """Manager for all available Studio entities.
+
+    Examples of Studio entities include Courses, Libraries, and Programs. All Studio
+    entities should implement `StudioEntity`.
+    """
+    NAMESPACE = 'openedx.studio_entity'
+
+    @classmethod
+    def get_enabled_entities(cls):
+        """Returns a list of enabled Studio entities."""
+        entities = cls.get_available_plugins()
+        enabled_entities = [entity for entity in entities.viewvalues() if entity.is_enabled()]
+
+        return enabled_entities
+
+
+class StudioEntity(object):
+    """Abstract class used to represent Studio entities.
+
+    Examples of Studio entities include Courses, Libraries, and Programs.
+    """
+    __metaclass__ = abc.ABCMeta
```

Comment:
No, I have no problem with abstract classes. The only thing I don't want is for the method of plugin to be "subclass this abstract class". I'd prefer to use entry_points and stevadore, rather than relying on particular code to be imported in order to register a plugin.